From 76d852d440cb2224867f0db92c64982003afe060 Mon Sep 17 00:00:00 2001
From: Denis Barbier <barbier@imacs.polytechnique.fr>
Date: Fri, 13 Apr 2018 15:45:23 +0200
Subject: [PATCH] Fix build with CMake 3.11

---
 python/src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/src/CMakeLists.txt b/python/src/CMakeLists.txt
index 7a2938c3b..969c26d55 100644
--- a/python/src/CMakeLists.txt
+++ b/python/src/CMakeLists.txt
@@ -23,7 +23,7 @@ macro (ot_add_python_module MODULENAME SOURCEFILE)
   ot_install_swig_file (${SOURCEFILE})
   if (${ARGC} GREATER 2)
     set (SWIG_MODULE_${MODULENAME}_EXTRA_DEPS ${ARGN})
-    list (APPEND SWIG_MODULE_${MODULENAME}_EXTRA_DEPS openturns/PythonWrappingFunctions.hxx)
+    list (APPEND SWIG_MODULE_${MODULENAME}_EXTRA_DEPS generate_docstrings openturns/PythonWrappingFunctions.hxx)
   endif ()
 
   set (swig_other_sources)
@@ -57,7 +57,7 @@ macro (ot_add_python_module MODULENAME SOURCEFILE)
     swig_add_library (${MODULENAME} LANGUAGE python SOURCES ${SOURCEFILE} ${swig_other_sources})
   endif ()
 
-  add_dependencies (${SWIG_MODULE_${MODULENAME}_REAL_NAME} generate_swig_runtime generate_docstrings)
+  add_dependencies (${SWIG_MODULE_${MODULENAME}_REAL_NAME} generate_swig_runtime)
   swig_link_libraries (${MODULENAME} OT)
 
   if (LINK_PYTHON_LIBRARY)
