diff --git a/Bonmin/examples/CppExample/MyBonmin.cpp b/Bonmin/examples/CppExample/MyBonmin.cpp
index 558bac5..0b7303f 100644
--- a/Bonmin/examples/CppExample/MyBonmin.cpp
+++ b/Bonmin/examples/CppExample/MyBonmin.cpp
@@ -69,6 +69,11 @@ int main (int argc, char *argv[])
     tminlp->printSolutionAtEndOfAlgorithm();
   }
 
+  // extra options
+  bonmin.options()->SetStringValue("honor_original_bounds", "yes");
+  bonmin.options()->SetNumericValue("constr_viol_tol", 1e-5);
+  bonmin.options()->SetNumericValue("bound_relax_factor", 0.0);
+
   //Now initialize from tminlp
   bonmin.initialize(GetRawPtr(tminlp));
 
