diff --git a/python/src/coupling_tools.py b/python/src/coupling_tools.py
index cb09978f6..2c1037bbb 100644
--- a/python/src/coupling_tools.py
+++ b/python/src/coupling_tools.py
@@ -159,7 +159,7 @@ class OTCalledProcessError(subprocess.CalledProcessError):
         err_msg = (
             (":\n" + self.stderr[:200].decode()) if self.stderr is not None else ""
         )
-        return super(OTCalledProcessError).__str__(self) + err_msg
+        return super(OTCalledProcessError, self).__str__() + err_msg
 
 
 def execute(
