diff --git a/python/test/t_Function_python.py b/python/test/t_Function_python.py
index 316fd5376..c904c67f3 100755
--- a/python/test/t_Function_python.py
+++ b/python/test/t_Function_python.py
@@ -96,15 +96,16 @@ myFunc = PythonFunction(2, 1, func_sample=a_exec_sample)
 outSample = myFunc(a_sample)
 print(outSample)
 
+cpus=-1 if sys.platform.startswith('linux') else None
 print('distributed exec only on a point')
 myFunc = PythonFunction(
-    2, 1, a_exec, n_cpus=None if sys.platform.startswith('win') else -1)
+    2, 1, a_exec, n_cpus=cpus)
 outSample = myFunc([100., 100.])
 print(outSample)
 
 print('distributed exec only on a sample')
 myFunc = PythonFunction(
-    2, 1, a_exec, n_cpus=None if sys.platform.startswith('win') else -1)
+    2, 1, a_exec, n_cpus=cpus)
 outSample = myFunc(a_sample)
 print(outSample)
 
