diff --git a/lib/test/t_Bonmin_std.cxx b/lib/test/t_Bonmin_std.cxx
index 5f818e28b..4dfa0a346 100644
--- a/lib/test/t_Bonmin_std.cxx
+++ b/lib/test/t_Bonmin_std.cxx
@@ -36,7 +36,7 @@ int main()
 {
   TESTPREAMBLE;
   OStream fullprint(std::cout);
-
+  Log::Show(Log::INFO);
   try
   {
     /* TEST ON SIMPLE FUNCTION */
@@ -85,7 +85,7 @@ int main()
     const Point referenceSolution = {1.0, 1.0, 0.5, 0.0};
 
     // Defining solvers to test
-    const Description solvers(Bonmin::GetAlgorithmNames());
+    const Description solvers({"B-BB"});
 
     for (UnsignedInteger i = 0; i < solvers.getSize(); ++i)
     {
